gensim/test/test_keyedvectors.py

```diff
@@ -400,6 +400,45 @@ def test_load_model_and_vocab_file_ignore(self):
             model.get_vector(u'どういたしまして'), np.array([.1, .2, .3], dtype=np.float32)))
 
 
+try:
+    import keras  # noqa: F401
+
+    keras_installed = True
+except ImportError:
+    keras_installed = False
+
+
+class WordEmbeddingsKeyedVectorsTest(unittest.TestCase):
+    def setUp(self):
+        self.vectors = EuclideanKeyedVectors.load_word2vec_format(
+            datapath('euclidean_vectors.bin'), binary=True, datatype=np.float64)
+
+    @unittest.skipIf(not keras_installed, 'keras needs to be installed for this test')
```

Comment:
Please move this decorator from the method to the class. If keras is not installed, none of the tests under the WordEmbeddingsKeyedVectorsTest should run, right?